packages/react-search-ui-views/src/view-helpers/getUrlSanitizer.js

```diff
@@ -0,0 +1,33 @@
+const VALID_PROTOCOLS = ["http:", "https:"];
+
+/**
+ *
+ * @param {URL} URLParser URL interface provided by browser https://developer.mozilla.org/en-US/docs/Web/API/URL
+ * @param {String} currentLocation String representation of the browser's current location
+ */
+export default function getUrlSanitizer(URLParser, currentLocation) {
+  // This function is curried so that dependencies can be injected and don't need to be mocked in tests.
+  return url => {
+    if (!url) return url;
+
+    let parsedUrl;
+
+    try {
+      // Attempts to parse a URL as absolute
+      parsedUrl = new URLParser(url);
+      // eslint-disable-next-line no-empty
```

Comment:
Ah I see you need both blocks to check for absolute AND relative urls (good call).  I still think there's probably a way to write this w/o the disables.

Reply:
That would change the behavior. I want it to try the absolute URL parsing, and if that fails, continue on to the second attempt using relative parsing.

Reply:
I guess I could do that in the second catch if you think it would be better 🤔

Reply:
Maybe I'm not understanding how try/catch works, but:

> I want it to try the absolute URL parsing, and if that fails, continue on to the second attempt using relative parsing.

Regardless of whether it fails or not, we're continuing on to relative parsing because we're not checking for `if parsedUrl` or anything in the 2nd try block.

Do we even need the first try/catch block in that case? What happens if we remove it?

Reply:
If the path is relative it will throw because it is an invalid URL

```
> const url = new URL('/relative-path')
> VM222:1 Uncaught TypeError: Failed to construct 'URL': Invalid URL
    at <anonymous>:1:13
```

I catch this so that doesn't get passed on to the user, I'd rather just attempt to parse this as a relative URL.

Does that make sense?

Reply:
Sorry this code is so confusing, I could separate those out into separate function if it helps readability.

Reply:
Sorry, my question was more along the lines of "can we just do `new URLParser(url, currentLocation)` to start with and skip the first `new URLParser(url)`?" - I def agree we shouldn't skip the 2nd parser with currentLocation.

Reply:
From what I can tell (from running the test for this file) nothing seems to break when removing lines 13-17.

I also tested this in the browser:
```
new URL('/foo/bar', 'https://github.com')
```
```
new URL('http://foo.bar', 'https://github.com')
```

Both seem to work just fine.